specs/StatusBarActivity.md

```diff
@@ -0,0 +1,155 @@
+<!-- USAGE
+  * Fill in each of the sections (like Background) below
+  * Wrap code with `single line of code` or ```code block```
+  * Before submitting, delete all <!-- TEMPLATE marked comments in this file,
+    and the following quote banner:
+-->
+# Background
+The browser has a Status bar that displays text when hovering over a link, or performing some activity. Currently, 
+developers are able to opt in to disable showing the status bar through the browser 
+settings.
+
+Developers would also like to be able to opt in to intercept the updates which would 
+normally be displayed by the Status bar, and show it using thier own custom UI. 
+# Description
+We propose two new events for WebView2 that would allow developers to 
+listen for Status bar updates which are triggered by activity on the WebView, and then handle those updates however they want in their applications.
+
+Developers will be able to: 
+
+1) Register an even handler for knowing when to show the status bar and what text to show
+2) Register an handler for knowing when the browser wants to dismiss the status bar
+
+
+# Examples
+## Win32 C++ Registering a listener for status bar showing
+```
+CHECK_FAILURE(m_webView->add_StatusBarShowing(
+    Callback<ICoreWebView2StatusBarShowingEventHandler>(
+        [this](ICoreWebView2* sender, ICoreWebView2StatusBarShowingEventArgs* args) -> HRESULT {
+
+            LPWSTR value;
+            CHECK_FAILURE(args->get_value(&value));
+            
+            // Handle status bar text in value
+
+            return S_OK;
+        }
+    ).Get(),
+&m_statusBarShowing));
+```
+## Win32 C++ Registering a listener for status bar hiding
+```
+CHECK_FAILURE(m_webView->add_StatusBarHiding(
+    Callback<ICoreWebView2StatusBarHidingEventHandler>(
+        [this](ICoreWebView2* sender) -> HRESULT {
+            
+            // Dismiss status bar
+
+            return S_OK;
+        }
+    ).Get(),
+&m_statusBarHiding));
+```
+## .NET / WinRT Registering a listener for status bar showing
+```
+webView.CoreWebView2.StatusBarShowing += (object sender, CoreWebView2StatusBarShowingEventArgs arg) =>
+{
+    string value = args.value;
+    // Handle status bar text in value
+};
+```
+## .Net/ WinRT Registering a listener for status bar hiding
+```
+webView.CoreWebView2.StatusBarHiding += (object sender) =>
+{
+     // Dismiss status bar
+};
+```
+# API Notes
+See [API Details](#api-details) Section below for API reference
+# API Details
+## Win32 C++
+```
+/*
+    Interface for the statusbar showing event args
+    The value property contains the status bar text
+*/
+[uuid(56acdbb8-ceac-11eb-b8bc-0242ac130003), object, pointer_default(unique)]
+interface ICoreWebView2StatusBarShowingEventArgs : IUnknown {
```

Comment:
(I'm not sure where our previous conversation on this went, so apologies for starting a new one rather than continuing the previous one)

Naming this StatusBarShowing is misleading. It doesn't sounds like the event is going to be raised repeatedly whenever the status bar message changes. I can't come up with a name that doesn't sound like we're merging two different events.

* Two events like `StatusBarVisibilityChanged`, `StatusBarMessageChanged` clearly describe what's happening.
* Alternatively, `StatusBarChanged` is less specific but is not misleading.

Reply:
@bradp0721 @champnic @jamesoli, @Tochibee says you had talked about this previously. Naming this StatusBarShowing is confusing. Any thoughts about these alternatives?

Reply:
I like 3), but was concerned if the "Dialog is hidden when message is empty" would pass API review. If you think it's ok, I think it's the simplest/clearest design.

Reply:
got it